cloudblue_connect.custom_adapter.rb

```diff
@@ -0,0 +1,659 @@
+                          label: 'Customer Country',
+                          hint: 'Provide the country of the related account, please use Alpha-2 code.',
+                          optional: false
+                        },
+
+                        {
+                          name: 'customer_contact_first_name',
+                          label: 'Customer Contact First Name',
+                          hint: 'Provide the contact name of the technical contact related with the account.',
+                          optional: false
+                        },
+
+                        {
+                          name: 'customer_contact_last_name',
+                          label: 'Customer Contact Last Name',
+                          hint: 'Provide the last name or family name of the technical contact related with the account.',
+                          optional: false
+                        },
+                        {
+                          name: 'customer_email',
+                          label: 'Customer Email',
+                          hint: "Provide the email address of the technical contact related with the account, this one typically is a requirement by vendors to process the request, additionally please note that this email will be used to send inquire requests related with the fulfillment of the request.",
+                          optional: false
+                        },
+                        {
+                          name: 'customer_phone_number',
+                          label: 'Customer Phone Number',
+                          hint: "Provide the phone number of the technical contact. We suggest to use international phone notation in order to ensure the correctness of the phone number.",
+                          optional: true
+                        },
+                        {
+                          name: 'product_id',
+                          label: 'Product ID',
+                          hint: "Provide the Product's ID.",
+                          optional: false
+                        },
+                        {
+                          name: 'product_items',
+                          label: 'Product Items',
+                          hint: '',
+                          optional: false,
+                          type: 'array', of: 'object', properties: [
+                            { optional: false, name: 'id', label: 'Item ID' },
+                            { optional: false, name: 'quantity', label: 'Quantity' }
+                          ]
+                        }
+
+                      ]
+                    end,
+
+      execute: lambda do |connection, input|
+        customer_external_uid = if input['customer_external_uid'].blank?
+                                  workato.uuid
+                                else
+                                  call(:get_tier_external_uid, connection, input['customer_external_id'])
+                                end
```

Comment:
Why is it this algorithm?

Correct algorithm seems to be
```suggestion
        customer_external_uid = if input['customer_external_uid'].blank?
                                  #if no UID in input, try to look for it by external ID (or autogenerate if not found)
                                  call(:get_tier_external_uid, connection, input['customer_external_id']) 
                                else
                                  #if UID is passed in the input explicitly, let's use it
                                  input['customer_external_uid']
                                end
```

Reply:
Yes that was a mistake, I made another commit that will fix this